config.m4

```diff
@@ -10,10 +10,11 @@ if test "$PHP_DDTRACE" != "no"; then
     PHP_SUBST(EXTRA_CFLAGS)
   fi
 
-  PHP_NEW_EXTENSION(ddtrace, src/ext/ddtrace.c src/ext/memory_limit.c src/ext/circuit_breaker.c src/ext/dispatch_setup.c src/ext/dispatch.c src/ext/request_hooks.c src/ext/compat_zend_string.c src/ext/dispatch_compat_php5.c src/ext/dispatch_compat_php7.c src/ext/backtrace.c src/ext/logging.c src/ext/env_config.c src/ext/serializer.c src/ext/mpack/mpack.c, $ext_shared,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -Wall -std=c11 -D_XOPEN_SOURCE=500)
+  PHP_NEW_EXTENSION(ddtrace, src/ext/ddtrace.c src/ext/memory_limit.c src/ext/circuit_breaker.c src/ext/dispatch_setup.c src/ext/dispatch.c src/ext/coms.c src/ext/coms_curl.c src/ext/coms_debug.c src/ext/request_hooks.c src/ext/compat_zend_string.c src/ext/dispatch_compat_php5.c src/ext/dispatch_compat_php7.c src/ext/backtrace.c src/ext/logging.c src/ext/env_config.c src/ext/serializer.c src/ext/mpack/mpack.c, $ext_shared,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -Wall -std=c11 -D_XOPEN_SOURCE=700)
```

Comment:
I forgot to mention that ever since we added the `-D_XOPEN_SOURCE=500` flag, the PHP tracer won't compile on masOS for PHP <= 7.1. But if I change the flag to just `-D_XOPEN_SOURCE` it compiles fine on macOS.

So could we move this flag to **env_config.h** (or where ever you think) and do some platform checks [similar to this](https://github.com/StrikeW/ads-demo/blob/97188250e013ad58ce5f6eb441733f45d4040d38/hiredis/fmacros.h#L13-L23)? :)

Reply:
I solved it by adding yet another file :)
without better naming ideas I named it `macros.h` - I think it handles our _XOPEN_SOURCE problem quite well.

Reply:
Nice! That's exactly what I would have named it! :D